extensions/wikia/AdEngine/js/provider/factoryWikiaGpt.js

```diff
@@ -1,21 +1,30 @@
 /*global define, require*/
 define('ext.wikia.adEngine.provider.factory.wikiaGpt', [
+	'ext.wikia.adEngine.adContext',
 	'ext.wikia.adEngine.adLogicPageParams',
 	'ext.wikia.adEngine.provider.gpt.helper',
-	'wikia.geo',
 	'wikia.log',
 	require.optional('ext.wikia.adEngine.lookup.services')
-], function (adLogicPageParams, gptHelper, geo, log, lookups) {
+], function (adContext, adLogicPageParams, gptHelper, log, lookups) {
 	'use strict';
-	var country = geo.getCountryCode();
 
-	function overrideSizes(slotMap, newSizes) {
-		var slotName;
-		for (slotName in slotMap) {
-			if (slotMap.hasOwnProperty(slotName) && newSizes[slotName]) {
-				slotMap[slotName].size = newSizes[slotName];
+	function overrideSizes(slotMap) {
+		var context = adContext.getContext();
+
+		if (context.opts.overrideLeaderboardSizes) {
+			for (var slotName in slotMap) {
+				if (slotMap.hasOwnProperty(slotName)) {
+					if (slotName.indexOf('TOP_LEADERBOARD') > -1) {
+						slotMap[slotName].size = '728x90';
+					}
+				}
 			}
 		}
+
+		if (context.opts.overridePrefootersSizes) {
+			slotMap.PREFOOTER_LEFT_BOXAD.size = '300x250,728x90,970x250';
```

Comment:
This is probably not for this story, but what do you guys think about keeping slot sizes in one place? I'm thinking about code readability. Sizes could be stored per slot, not per-provider. Current situation is that size-information is split into different places in codebase (different providers) and then more complicated by "overrideSizes" function. @nandy-andy @drets @fraszczakszymon